### ISSUE TITLE
The training has been stuck at 0%

### ISSUE THREAD

**Issue by linxiang4200** [guoyww/AnimateDiff#201]
I don't know the reason. After starting the training command, the progress has remained at 0%, and there are no error messages. Has anyone else encountered this issue? The output is as follows:

torchrun --nnodes=1 --nproc_per_node=1 train.py --config configs/training/training.yaml
loaded temporal unet's pretrained weights from /home/work/stable-diffusion-v1-5/unet ...
### Temporal Module Parameters: 417.1376 M
10/24/2023 19:28:13 - INFO - root - ***** Running training *****
10/24/2023 19:28:13 - INFO - root -   Num examples = 836
10/24/2023 19:28:13 - INFO - root -   Num Epochs = 1
10/24/2023 19:28:13 - INFO - root -   Instantaneous batch size per device = 4
10/24/2023 19:28:13 - INFO - root -   Total train batch size (w. parallel, distributed & accumulation) = 4
10/24/2023 19:28:13 - INFO - root -   Gradient Accumulation steps = 1
10/24/2023 19:28:13 - INFO - root -   Total optimization steps = 100
Steps:   0%|                                                                                                                                                                      | 0/100 [00:00<?, ?it/s]

**Comment by xiaolxl111** [comment]
I have also encountered this problem. Have you resolved it?

**Comment by linxiang4200** [comment]
> I have also encountered this problem. Have you resolved it?

I encountered this issue because I forgot to set the title row in the CSV file. Could you check if the first row of your CSV file is the title row?

**Comment by xiaolxl111** [comment]
Thank you for your help. There is indeed a title error in my CSV file

**Comment by JoshonSmith** [comment]
hank you for your help. There is indeed a title error in my CSV file.
my csv file contains two column：name,contentUrl.

**Comment by AlonzoLeeeooo** [comment]
![image](https://github.com/guoyww/AnimateDiff/assets/68578066/c9dec87a-57ea-483e-9e4f-98387aebfd31)
Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻

**Comment by linxiang4200** [comment]
> ![image](https://private-user-images.githubusercontent.com/68578066/335094828-c9dec87a-57ea-483e-9e4f-98387aebfd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjUyNzksIm5iZiI6MTcxNzEyNDk3OSwicGF0aCI6Ii82ODU3ODA2Ni8zMzUwOTQ4MjgtYzlkZWM4N2EtNTdlYS00ODNlLTllNGYtOTgzODdhZWJmZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NGRjMGJiMzVmZmQ0MDlkOThjYmQ4NWYzZTAzMDkzMjAxOGFmYjlkNmZiNTY0ODc2MGNiZmMzMzAzN2JmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxJg0oHMSiim_7wI04Bk7TByC0fpW4bQdDWSFnieA6I) Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻

The first three columns should be videoid, name, page_dir

https://github.com/guoyww/AnimateDiff/blob/cf80ddeb47b69cf0b16f225800de081d486d7f21/animatediff/data/dataset.py#L41

**Comment by AlonzoLeeeooo** [comment]
> > ![image](https://private-user-images.githubusercontent.com/68578066/335094828-c9dec87a-57ea-483e-9e4f-98387aebfd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjUyNzksIm5iZiI6MTcxNzEyNDk3OSwicGF0aCI6Ii82ODU3ODA2Ni8zMzUwOTQ4MjgtYzlkZWM4N2EtNTdlYS00ODNlLTllNGYtOTgzODdhZWJmZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NGRjMGJiMzVmZmQ0MDlkOThjYmQ4NWYzZTAzMDkzMjAxOGFmYjlkNmZiNTY0ODc2MGNiZmMzMzAzN2JmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxJg0oHMSiim_7wI04Bk7TByC0fpW4bQdDWSFnieA6I) Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻
> 
> The first three columns should be videoid, name, page_dir
> 
> https://github.com/guoyww/AnimateDiff/blob/cf80ddeb47b69cf0b16f225800de081d486d7f21/animatediff/data/dataset.py#L41

Thanks for your fast responding! I will try and update with you.

Best,

**Comment by Ghy0501** [comment]
> > > ![image](https://private-user-images.githubusercontent.com/68578066/335094828-c9dec87a-57ea-483e-9e4f-98387aebfd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjUyNzksIm5iZiI6MTcxNzEyNDk3OSwicGF0aCI6Ii82ODU3ODA2Ni8zMzUwOTQ4MjgtYzlkZWM4N2EtNTdlYS00ODNlLTllNGYtOTgzODdhZWJmZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NGRjMGJiMzVmZmQ0MDlkOThjYmQ4NWYzZTAzMDkzMjAxOGFmYjlkNmZiNTY0ODc2MGNiZmMzMzAzN2JmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxJg0oHMSiim_7wI04Bk7TByC0fpW4bQdDWSFnieA6I) Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻
> > 
> > 
> > The first three columns should be videoid, name, page_dir
> > https://github.com/guoyww/AnimateDiff/blob/cf80ddeb47b69cf0b16f225800de081d486d7f21/animatediff/data/dataset.py#L41
> 
> Thanks for your fast responding! I will try and update with you.
> 
> Best

Hi, I don't do any additional processing on dataset.py or .csv, but it is able to run image_fintune.yaml and training.yaml, just had to wait for a while(~2-3min). However, I think the output of the training is strange, I get a motion_module of 4.76GB while the provided motion_module is around 1.6GB.

**Comment by AlonzoLeeeooo** [comment]
> > > > ![image](https://private-user-images.githubusercontent.com/68578066/335094828-c9dec87a-57ea-483e-9e4f-98387aebfd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjUyNzksIm5iZiI6MTcxNzEyNDk3OSwicGF0aCI6Ii82ODU3ODA2Ni8zMzUwOTQ4MjgtYzlkZWM4N2EtNTdlYS00ODNlLTllNGYtOTgzODdhZWJmZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NGRjMGJiMzVmZmQ0MDlkOThjYmQ4NWYzZTAzMDkzMjAxOGFmYjlkNmZiNTY0ODc2MGNiZmMzMzAzN2JmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxJg0oHMSiim_7wI04Bk7TByC0fpW4bQdDWSFnieA6I) Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻
> > > 
> > > 
> > > The first three columns should be videoid, name, page_dir
> > > https://github.com/guoyww/AnimateDiff/blob/cf80ddeb47b69cf0b16f225800de081d486d7f21/animatediff/data/dataset.py#L41
> > 
> > 
> > Thanks for your fast responding! I will try and update with you.
> > Best
> 
> Hi, I don't do any additional processing on dataset.py or .csv, but it is able to run image_fintune.yaml and training.yaml, just had to wait for a while(~2-3min). However, I think the output of the training is strange, I get a motion_module of 4.76GB while the provided motion_module is around 1.6GB.

It seems like the storage equals to the ones of a total of U-net (~3GB) and motion modules (~1.6GB). Maybe you can have a sanity check about whether the model weights of U-net is frozen.

**Comment by AlonzoLeeeooo** [comment]
> > ![image](https://private-user-images.githubusercontent.com/68578066/335094828-c9dec87a-57ea-483e-9e4f-98387aebfd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjUyNzksIm5iZiI6MTcxNzEyNDk3OSwicGF0aCI6Ii82ODU3ODA2Ni8zMzUwOTQ4MjgtYzlkZWM4N2EtNTdlYS00ODNlLTllNGYtOTgzODdhZWJmZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMDkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NGRjMGJiMzVmZmQ0MDlkOThjYmQ4NWYzZTAzMDkzMjAxOGFmYjlkNmZiNTY0ODc2MGNiZmMzMzAzN2JmMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxJg0oHMSiim_7wI04Bk7TByC0fpW4bQdDWSFnieA6I) Hi @linxiang4200, could you please help me to check my csv file? I have uploaded a screenshot of the csv file. Is there any problem with this csv file? What should I do to fix the problems? Please help 👍🏻
> 
> The first three columns should be videoid, name, page_dir
> 
> https://github.com/guoyww/AnimateDiff/blob/cf80ddeb47b69cf0b16f225800de081d486d7f21/animatediff/data/dataset.py#L41

Hi @linxiang4200 ,
The first three column does not seem to be the problem. We can look at the code specifically. In `line 25` of `dataset.py`, the csv file is loaded via:
 ```python
self.dataset = list(csv.DictReader(csvfile))
```
Herein, `self.dataset` is a list with all the data samples, where each sample is a dict with the following keys: `videoid`, `contentUrl`, `duration`, `page_dir`, `name`. So it is feasible to directly load `videoid`, `name`, and `page_dir` from each sample. There is no need to re-construct the csv file format.

I have also tried another way to load data, where I pre-extract all the captions from the csv file and save them as `.txt` files locally. Then in `dataset.py`, the videos and captions are directly loaded from local folders, instead of using the csv file. **The code still gets stuck at 0%.** Therefore, I am not whether the csv file is where the problem lies actually.

I guess that the problem might be caused by the use of `decord` to load videos, I've heard that `decord` requires large CPU resources to load `.mp4` files. Currently I am trying to split videos into individual frames, and load the splitted frames with image IO. Hopefully this might address the problem.

If you have any idea in mind of the aforementioned findings, please feel free to discuss with me.

Best
